### PR TITLE
Hook debounce implementation

### PR DESCRIPTION
**Pull request by CristianRWolox**
## Summary

The hook was implemented based on the useDebounce osde-on project

## Screenshots

recipe

## Trello Card
[Card](https://trello.com/c/ov1cZcos/214-hook-usedebounce)

### CODE REVIEW

File: cookbook-react-native/recipes/hooks/debounce/useDebounce.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { useEffect, useState } from 'react';
+import { Nullable } from '@interfaces/globalInterfaces';
+
+const DEFAULT_DELAY = 500;
```
Comment (guidoprinc) [inline cookbook-react-native/recipes/hooks/debounce/useDebounce.ts:4]: this is half a second, i would increase the default delay

Comment (CristianRWolox) [reply cookbook-react-native/recipes/hooks/debounce/useDebounce.ts:4]: Do you think 750 would be more suitable?

Comment (guidoprinc) [reply cookbook-react-native/recipes/hooks/debounce/useDebounce.ts]: i would say a whole second

Comment (CristianRWolox) [reply cookbook-react-native/recipes/hooks/debounce/useDebounce.ts]: Done!

**Review by guidoprinc — APPROVED** [review]

**Pull request merged**